### PR TITLE
Abort a hot reload if we detect an error in flutter.yaml

### LINKED ISSUE #6726: Hot reload doesn't stop the reload if it hits an error in the flutter.yaml file
It seems like it should, because otherwise the error message might get buried in other log spam, such as for example the corresponding error when trying to load the asset that the author failed to correctly mention in the flutter.yaml file.

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Better handling of Flutter.yaml errors.
- [x] Do not allow a reload to happen with an error in flutter.yaml

Fixes https://github.com/flutter/flutter/issues/6726

### CODE REVIEW

File: packages/flutter_tools/lib/src/asset.dart [hunk 1]
```diff
@@ -449,10 +456,10 @@ Future<int> _validateFlutterYamlManifest(Object manifest) async {
   if (validator.validate(manifest)) {
     return 0;
   } else {
+    printStatus('Error detected in flutter.yaml:', emphasis: true);
     if (validator.errors.length == 1) {
-      printError('Error in flutter.yaml: ${validator.errors.first}');
+      printError('${validator.errors.first}');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/asset.dart]: though really the multiline version seems fine to always use, imho

File: packages/flutter_tools/lib/src/hot.dart [hunk 2]
```diff
@@ -387,12 +387,15 @@ class HotRunner extends ResidentRunner {
...
       return false;
     }
-    Status devFSStatus = logger.startProgress('Syncing files to device...');
     final bool rebuildBundle = bundle.needsBuild();
     if (rebuildBundle) {
       printTrace('Updating assets');
-      await bundle.build();
+      int result = await bundle.build();
+      if (result != 0) {
+        return false;
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/hot.dart]: nit: remove braces

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/hot.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/hot.dart]: Done.

File: packages/flutter_tools/lib/src/asset.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/asset.dart]: Always multiline

**Pull request merged**